rmw_fastrtps_shared_cpp/src/participant.cpp

```diff
@@ -194,6 +197,20 @@ rmw_fastrtps_shared_cpp::create_participant(
   }
   if (env_value != nullptr) {
     leave_middleware_default_qos = strcmp(env_value, "1") == 0;
+  } else if (env_value == nullptr || !leave_middleware_default_qos) {
+    error_str = rcutils_get_env("ROS_PUBLICATION_MODE", &env_value);
+    if (error_str != NULL) {
+      RCUTILS_LOG_DEBUG_NAMED("rmw_fastrtps_shared_cpp", "Error getting env var: %s\n", error_str);
+      return nullptr;
+    }
+    if (env_value != nullptr) {
+      // Synchronous publishing
+      if (strcmp(env_value, "SYNCHRONOUS") == 0) {
+        publishing_mode = publishing_mode_t::SYNCHRONOUS;
+      } else if (strcmp(env_value, "AUTO") == 0) {
+        publishing_mode = publishing_mode_t::AUTO;
+      }
```

Comment:
@JLBuenoLopez-eProsima if it's neither, a warning log stating that the given value is unknown and the default will take over would be nice.

Reply:
Done